### ISSUE TITLE
CreateIdAttr() always creates empty info:id token

### ISSUE THREAD

**Issue by shauser67** [PixarAnimationStudios/USD#1681]
### Description of Issue

After switching from an older version to 21.11 I am getting this error message when creating an info::id.

> Coding Error: in _SetValueImpl at line 5996 of /home/sh/dev/USD-21.11/pxr/usd/usd/stage.cpp -- Type mismatch for </test_export/mesh_test_export_000/material/shader.info:id>: expected 'TfToken', got ''

**C++ code**

```
auto usd_shader = pxr::UsdShadeShader::Define (usd_stage, usd_shader_path);

usd_shader.CreateIdAttr (pxr::VtValue (pxr::TfToken ("UsdPreviewSurface")));
```

**Generated USD file**

```
def Mesh "mesh_test_export_000"
{
    [...]

    def Material "material"
    {
        def Shader "shader"
        {
            uniform token info:id

            [...]
        }
    }
}
```

Inspired by other USD exporters I tried to use a private token list but it shows the same error message and info::id is still empty.

**C++ code**
```
PXR_NAMESPACE_OPEN_SCOPE

TF_DEFINE_PRIVATE_TOKENS(
    _tokens,
    (UsdPreviewSurface)
    (UsdUVTexture)
    (UsdPrimvarReader_float2)
);

PXR_NAMESPACE_CLOSE_SCOPE

auto usd_shader = pxr::UsdShadeShader::Define (usd_stage, usd_shader_path);

usd_shader.CreateIdAttr (pxr::VtValue (pxr::_tokens->UsdPreviewSurface));
```

Is there another way to create proper info::ids?

### System Information (OS, Hardware)
Linux / Ubuntu 21.01

### Package Versions
USD 21.11

### Build Flags
Built without imaging and python

**Comment by spiffmon** [comment]
Hi @shauser67 , that error indicates a failure of RTTI comparisons.  You can search the usd-interest forum to see some issues folks have had in this area.  We don't use the build configuration you posted, so I'm unsure if there could possibly be something about it that is causing this.  Are you setting any other build flags?  If you try a fuller build, does the problem persist?

**Comment by shauser67** [comment]
Hi @spiffmon, thanks for the advice. I tried several build configurations and finally could solve the issue. Instead of removing imaging and python from USD this pretty basic build configuration worked for me:

`python3 build_scripts/build_usd.py --embree --opencolorio --openimageio build`

**Comment by meshula** [comment]
@shauser67 what is the simplest build configuration you tried, that did not work? I would like to get some clues about which of the optional components might have introduced the problem?

**Comment by shauser67** [comment]
@meshula I tested several build configurations (--no-python / --no-usdview --no-python --no-tools) but couldn't replicate the initial problem. All worked well.

Finally I figured out what caused the issue. I initially built two versions of USD, one without python and tools for inclusion in the application, and one with python and tools for testing the generated output. The application was linked against the first version but uses the runtime libraries of the second version. With a single USD build for both the application and testing all works well.

Sorry for the confusion!
